File: apps/bf/native/ios/tbd_native/GamesLobby/GamesLobbyObject.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(GamesLobbyObject, NSObject)

+ (BOOL)requiresMainQueueSetup {
  return YES;
}

RCT_EXTERN_METHOD(triggerConfirmationPopUp:(BOOL *) value)
RCT_EXTERN_METHOD(gamesLobbyTabActive:(BOOL *) value)
RCT_EXTERN_METHOD(closeModalView:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(openWebView:(NSString *)url
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: apps/pp/native/ios/tbd_native/GamesLobby/GamesLobbyObject.swift
import UIKit
import Foundation
import GamesFramework

@objc(GamesLobbyObject)
@MainActor
class GamesLobbyObject: NSObject {
    let gamesLobby = GamesLobbyView()
  
    @objc func triggerConfirmationPopUp(_ value : Bool) {
        gamesLobby.triggerConfirmation = value
    }
  
    @objc func gamesLobbyTabActive(_ value : Bool) {
        gamesLobby.gamesLobbyTabActive = value
    }
  
    @objc func closeModalView(_ resolver: @escaping RCTPromiseResolveBlock,
                              rejecter: @escaping RCTPromiseRejectBlock) {
        gamesLobby.closeModalView(resolver: resolver)
    }
  
    @objc func openWebView(_ url: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        gamesLobby.openWebView(url, resolver: resolver)
    }
}
